interview-questions.md

```diff
@@ -14,44 +14,51 @@ Researched answer:
+Researched answer: You cannot use github without git, but you can use git without github. While this is obvious now that I read it it never occurred to me before.
 
 2. Which JavaScript operators will return a Boolean value?
 
-Your answer:
+Your answer: Relational operators will return a boolean. Those are the greater than/less than and comparison operators.
 
-Researched answer:
+Researched answer: The algebraic boolean operators will also return boolean values. Those are &&, ||, and the BANG! operator.
 
 1. What is the difference between an array's index and its value?
 
-Your answer:
+Your answer: Every element of every javascript array has an index. These indexes start at 0 and increment by 1 as elements are added to the array. These elements are the value. Arrays can hold as values; any primitive data type, other arrays, objects, and more.
 
-Researched answer:
+Researched answer: You can also have mixed type arrays. Meaning you can even have array values of more than one data type at a time. However the indexes will also be numbers starting from 0, incrementing by 1.
 
 4. Compare and contrast arrays and strings. Please include ways in which they are the same and ways in which they are different.
 
-Your answer:
+Your answer:Arrays and Strings are both collections of smaller elements. An array being made up individual values that can house all types. Strings are simply a collection of characters. Both are 0 indexed and have access to some of the same methods, such as .length.
 
-Researched answer:
+Researched answer: Arrays are considered objects, which means they can only ever be strictly equal to themselves. This is a stark contrast the primitive data type of string.
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-Your answer:
 
-Researched answer:
+Your answer:Pair programming is the process of having two people work on the same file. One programmer operates as the driver, sharing their screen and typing in the target file. The other programmer takes on the role of navigator. They simply share the mental load, assisting the driver with completion of the tasks. Driver types, Navigator researches.
+
+Researched answer: To be successful with pair programming actually invovles a high skill level from both members. The partners must have the soft skills to collaborate effectivly and stay engaged, while also having the hard skills necessary to write effective code.
 
 ## Looking Ahead: Terms for Next Week
 
 Research and define the following terms to the best of your ability.
 
-1. Higher Order Functions:
+1. Higher Order Functions: A higher order function is a function that either accepts as an argument, or returns another function. 
+
+After some time to mull over the concept, I realized that .sort is a higher order function(I think), because it accepts a function as an argument. However, I still have no meaningful understanding of how to write my own higher order functions.
```

Comment:
To write your own higher order function, it would involve creating a normal function, and then a separate function that will take that function as an argument. However, there are many higher order functions already made by other developers to perform a wide variety of functionalities, packaged for easy access (such as .map). `.sort` is not a higher order function, but a built-in method.